streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSuppressSupplier.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.internals.suppress.KTableSuppressProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+public class KTableSuppressSupplier<K, V> implements KTableProcessorSupplier<K, V, V>, ProcessorSupplier<K, Change<V>> {
+
+    private final String storeName;
+    private final KTableSuppressProcessor<K, V> suppressProcessor;
+
+    public KTableSuppressSupplier(final KTableSuppressProcessor<K, V> suppressProcessor, final String storeName) {
+        this.storeName = storeName;
+        this.suppressProcessor = suppressProcessor;
+    }
+
+    @Override
+    public KTableValueGetterSupplier<K, V> view() {
+        return new KTableValueGetterSupplier<K, V>() {
+
+            public KTableValueGetter<K, V> get() {
+                return new KTableSuppressValueGetter();
+            }
+
+            @Override
+            public String[] storeNames() {
+                return new String[]{storeName};
+            }
+        };
+    }
+
+    @Override
+    public void enableSendingOldValues() {
+        throw new UnsupportedOperationException("enableSendingOldValues is not supported by KTableSuppressSupplier");
```

Comment:
Not clear why we decide to throw in `enableSendingOldValues` while trying to handle `view` gracefully? It seems to me that we should handle both cases otherwise it will still throw and fail right?

Reply:
Yeah, I think that there's a larger "lookback" feature that I wasn't aware of when I implemented Suppress. It seems like it needs a first-class solution, and probably just mixing in this interface would just surface a different exception at run time.

I'm not sure without spending some time to look at it, but it seems we need to enable "old values" upstream and then add the ability to store the old values as well. Actually, this may already be partially supported, with the FullChangeSerde.

The other missing API is the valuegetter. We might need to actually implement that, acting as a cache (look in the buffer first, then look upstream), or, since we know the buffer _always_ reflects the upstream state anyway, we can just directly look upstream.

Reply:
```
or, since we know the buffer always reflects the upstream state anyway, we can just directly look upstream.
```

Yeah I think this would work, and it is the simplest approach as well (the simpler, the better).

I feel it is a pretty critical bug fix to get in asap since it may affect quite lot of DSL cases, right?

Reply:
yes, I think so.

Reply:
@ouertani given this is an important fix to have, do you have time for addressing the above comments so we can merge before 2.3 release, which is very near?

Basically we need to handle `sendOldValues` and `view` gracefully than throwing unsupported exceptions.